model/host/host.go

```diff
@@ -1108,9 +1118,134 @@ func (hosts HostGroup) GetHostIds() []string {
 	return ids
 }
 
+// getNumContainersOnParents returns a slice of uphost parents and their respective
+// number of current containers currently running in order of longest expected
+// finish time
+func GetNumContainersOnParents(d distro.Distro) ([]ContainersOnParents, error) {
+	allParents, err := findUphostParentsByContainerPool(d.ContainerPool)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not find running parent hosts")
+	}
+
+	numContainersOnParents := make([]ContainersOnParents, 0)
+	// parents come in sorted order from soonest to latest expected finish time
+	for i := len(allParents) - 1; i >= 0; i-- {
+		parent := allParents[i]
+		currentContainers := []Host{}
+		if parent.Status == evergreen.HostRunning {
+			currentContainers, err = parent.GetContainers()
+			if err != nil {
+				return nil, errors.Wrapf(err, "Could not find containers for parent %s", parent.Id)
+			}
+		}
+		if len(currentContainers) < parent.ContainerPoolSettings.MaxContainers {
+			numContainersOnParents = append(numContainersOnParents,
+				ContainersOnParents{
+					ParentHost:    parent,
+					NumContainers: len(currentContainers),
+				})
+		}
+	}
```

Comment:
is this the kind of thing that could be pushed up to the database in an aggregation?

Reply:
I don't know a lot about aggregation (do you mean to get the list of ContainersOnParents via an aggregation?), but this function was just moved from the scheduler. Unless you have an idea how to do it, it might be best to make that a separate ticket.